source_http.go

```diff
@@ -33,14 +34,26 @@ func (s *HttpImageSource) GetImage(req *http.Request) ([]byte, error) {
 }
 
 func (s *HttpImageSource) fetchImage(url *url.URL, ireq *http.Request) ([]byte, error) {
-	req := newHTTPRequest(url)
+	// Check remote image size by fetching HTTP Headers
+	if s.Config.MaxAllowedSize > 0 {
+		req := newHTTPRequest(s, ireq, "HEAD", url)
+		res, err := http.DefaultClient.Do(req)
+		if err != nil {
+			return nil, fmt.Errorf("Error fetching image http headers: %v", err)
+		}
+		defer res.Body.Close()
```

Comment:
I think we can actually close the Body contents reader without deferring it, since we won't consume it:
```go
res.Body.Close()
```